assets/src/stories-editor/components/higher-order/with-snap-targets.js

```diff
@@ -0,0 +1,182 @@
+	 * @return {Function} Proxy setter.
+	 */
+	const getSetter = ( lowerBound, upperBound ) => {
+		return ( obj, prop, value ) => {
+			prop = Math.round( prop );
+
+			if ( prop < lowerBound || prop > upperBound ) {
+				return true;
+			}
+
+			const hasSnapLine = ( item ) => obj[ prop ].find( ( snapLine ) => isShallowEqual( item[ 0 ], snapLine[ 0 ] ) && isShallowEqual( item[ 1 ], snapLine[ 1 ] ) );
+
+			if ( obj.hasOwnProperty( prop ) && ! hasSnapLine( value ) ) {
+				obj[ prop ].push( value );
+			} else {
+				obj[ prop ] = [ value ];
+			}
+
+			obj[ prop ] = obj[ prop ].sort();
+
+			return true;
+		};
+	};
+
+	return {
+		/**
+		 * Horizontal snap function.
+		 *
+		 * @param {number} targetTop The top position of the currently dragged/resized block.
+		 * @param {number} targetBottom The bottom position of the currently dragged/resized block.
+		 * @return {Object.<number,Array.<Array.<number, number>>>} Dictionary with horizontal snap targets.
+		 */
+		horizontalSnaps: ( targetTop, targetBottom ) => {
+			const snaps = new Proxy( {
+				// Left page border.
+				0: [ getVerticalLine( 0 ) ],
+				// Center of the page.
+				[ STORY_PAGE_INNER_WIDTH / 2 ]: [ getVerticalLine( STORY_PAGE_INNER_WIDTH / 2 ) ],
+				// Right page border.
+				[ STORY_PAGE_INNER_WIDTH ]: [ getVerticalLine( STORY_PAGE_INNER_WIDTH ) ],
+			},
+			{
+				set: getSetter( 0, STORY_PAGE_INNER_WIDTH ),
+			} );
+
+			for ( const blockElement of siblings ) {
+				const { top, right, bottom, left } = getRelativeElementPosition( blockElement, parentBlockElement );
+				const center = left + ( ( right - left ) / 2 );
+
+				const start = targetTop < top ? targetTop : top;
+				const end = targetBottom > bottom ? targetBottom : bottom;
+
+				snaps[ left ] = getVerticalLine( left, start, end );
+				snaps[ right ] = getVerticalLine( right, start, end );
+				snaps[ center ] = getVerticalLine( center, start, end );
+			}
+
+			return snaps;
+		},
+
+		/**
+		 * Vertical snap function.
+		 *
+		 * @param {number} targetLeft The left position of the currently dragged/resized block.
+		 * @param {number} targetRight The right position of the currently dragged/resized block.
+		 * @return {Object.<number,Array.<Array.<number, number>>>} Dictionary with vertical snap targets.
+		 */
+		verticalSnaps: ( targetLeft, targetRight ) => {
+			const snaps = new Proxy( {
+				// Top page border.
+				0: [ getHorizontalLine( 0 ) ],
+				// Center of the page.
+				[ STORY_PAGE_INNER_HEIGHT / 2 ]: [ getHorizontalLine( STORY_PAGE_INNER_HEIGHT / 2 ) ],
+				// Bottom page border.
+				[ STORY_PAGE_INNER_HEIGHT ]: [ getHorizontalLine( STORY_PAGE_INNER_HEIGHT ) ],
+			},
+			{
+				set: getSetter( 0, STORY_PAGE_INNER_HEIGHT ),
+			} );
+
+			for ( const blockElement of siblings ) {
+				const { top, right, bottom, left } = getRelativeElementPosition( blockElement, parentBlockElement );
+				const center = top + ( ( bottom - top ) / 2 );
```

Comment:
Less obvious, but simpler - it is just finding the average 🤷‍♂

```suggestion
				const center = ( top + bottom ) / 2;
```

Reply:
Not sure what I was thinking there 🤔 

Will update all the occurrences.